crates/sui/src/upgrade_compatibility.rs

```diff
@@ -1250,13 +1188,36 @@ fn format_param(
         Type::Reference(t) => {
             format!("&{}", format_param(t, type_params, secondary)?)
         }
+        Type::Struct {
+            address,
+            module,
+            name,
+            type_arguments,
+            ..
+        } => {
+            if type_arguments.len() > 0 {
+                format!(
+                    "{}::{}::{}<{}>",
+                    address.to_hex_literal(),
+                    module,
+                    name,
+                    type_arguments
+                        .iter()
+                        .map(|t| format_param(t, type_params.clone(), secondary))
+                        .collect::<Result<Vec<_>, _>>()?
+                        .join(", ")
+                )
+            } else {
+                format!("{}", param)
+            }
+        }
         _ => format!("{}", param),
     })
 }
 
-/// Return a diagnostic for a function signature mismatch.
-/// Start by checking the lengths of the parameters and returns and return a diagnostic if they are different.
-/// If the lengths are the same check each parameter piece wise and return a diagnostic for each mismatch.
+/// Returns diagnostics for each signature mismatch in the given function.
+/// Start by checking the lengths of the parameters and returns and return add a diagnostic if they are different.
+/// If the lengths are the same check each parameter piece wise and return add a diagnostic for each mismatch.
```

Comment:
nit: this doesn't make grammatical sense (maybe you want to remove the `return`?)